testing/README.md

```diff
@@ -0,0 +1,390 @@
+        sinon.stub(subject, 'myBadFunc').returns(true);
+        assert.false(subject.doSomethingElse())
+      });
+    });
+
+    // good
+    suite('#myBadFunc', () => {
+      let subject;
+
+      setup(() => {
+        subject = new MyComponent();
+        sinon.stub(subject, 'myBadFunc').returns(true);
+      });
+
+      test('it does something', () => {
+        assert.true(subject.doSomething())
+      });
+
+      test('it does something else', () => {
+        assert.false(subject.doSomethingElse())
+      });
+    });
+  });
+  ```
+
+- Something something about how to name your tests
+
+- Do not rely on test ordering for your tests to run correctly. Imagine each test was going to be run in a separate process, and must therefore entirely setup any preconditions before running.
+
+  > Why? Tests that depend on anything outside of them will inevitably break as more tests are added. Other developers will assume there is no shared context between tests; don’t violate that assumption!
+
+- Use sinon’s spies/ stubs instead of rolling your own version.
+
+  > Why? Spies and stubs provide a consistent interface in case the tests need to be changed, reduce boilerplate so you can focus on the subject under test, can easily be restored to their original versions, and can be tested with more meaningful assertions.
+
+  ```js
+  suite('MyComponent', () => {
+    // bad
+    suite('#myBadFunc', () => {
+      let original;
+      let calledWith;
+
+      setup(() => {
+        spy = function(firstArg) { calledWith = firtArg; }
+        let original = subject.myBadFunc;
+        subject.myBadFunc = spy;
+      });
+
+      teardown(() => {
+        subject.myBadFunc = original;
+      });
+
+      test('it works', () => {
+        subject.doSomething();
+        assert.equal(calledWith, 'some arg');
+      });
+    });
+
+    // good
+    suite('#myGoodFunc', () => {
+      let spy;
+
+      setup(() => {
+        spy = sinon.spy(subject, 'myGoodFunc');
+      });
+
+      teardown(() => {
+        spy.restore();
+      });
+
+      test('it works', () => {
+        subject.doSomething();
+        assert.calledWith(spy, 'some arg');
+      });
+    });
+  });
+  ```
+
+- Use the most meaningful assertion for what you are trying to test. For a list of available assertions, make sure to check out [chai’s `assert` docs](http://chaijs.com/api/assert/).
+
+  > Why? More specific assertions have more specific error messages, making it easier for you (and any developer that deals with this code in the future) to track down and fix test failures.
+
+  ```js
+  // bad
+  assert.isTrue(foo === bar);
+
+  // good
+  assert.equal(foo, bar);
+  ```
+
+- Use sinon’s custom spy assertions, exposed to chai’s `assert` object.
+
+  ```js
+  // somewhere in the test setup...
+  // (see http://sinonjs.org/docs/#assert-expose)
+  sinon.assert.expose(assert);
+
+  // bad
+  assert.isTrue(spy.called);
+
+  // good
+  assert.called(spy);
```

Comment:
Another good option is testing `callCount`, this can help catch situations where functions are accidentally invoked multiple times.